.test/tests/valkey-integration-tests/run.sh

```diff
@@ -0,0 +1,253 @@
+#!/bin/bash
+set -euo pipefail
+
+image="$1"
+CONTAINER_NAME="valkey-test-$(date +%s)-$$"
+test_results=()
+TEST_FRAMEWORK_REPO="https://github.com/valkey-io/valkey-test-framework.git"
+
+cleanup_container() {
+    if docker ps -q -f name="$CONTAINER_NAME" | grep -q .; then
+        docker stop "$CONTAINER_NAME" >/dev/null 2>&1
+        docker rm "$CONTAINER_NAME" >/dev/null 2>&1
+    fi
+}
+
+summary() {
+    local exit_code=$?
+
+    cleanup_container
+    
+    if [ ${#test_results[@]} -gt 0 ]; then
+        echo "=== Test Results Summary ==="
+        printf '%s\n' "${test_results[@]}"
+    fi
+    
+    exit $exit_code
+}
+
+setup_test_framework() {
+    local dir="$1"
+    if [ ! -d "$dir" ]; then
+        git clone "$TEST_FRAMEWORK_REPO"
```

Comment:
```suggestion
        git clone "$TEST_FRAMEWORK_REPO" valkey-test-framework
```

This makes it easy for the reader, when you use `valkey-test-framework` below